crates/storage/src/schema/revision_0055.rs

```diff
@@ -0,0 +1,26 @@
+use anyhow::Context;
+
+pub(crate) fn migrate(tx: &rusqlite::Transaction<'_>) -> anyhow::Result<()> {
+    tx.execute_batch(
```

Comment:
I think you should also add some cleanup code to `Transaction::purge_block()` that removes block-related data in case of a reorg. That's another case where I think that `block_number` in these tables should refer to the block number that removed these indices instead of the block number where we're expecting these to be removed.